text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+- Feature Name: Dataset Push & Pull
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Replace `publish/add` with `push/pull`, which default to pushing as much data as possible, and pulling only the latest version. Implient _automatic following_ within `checkout` to get git-clone like behaviour. Introduce _retention strategies_, a subject for later RFCs.
+
+# Motivation
+[motivation]: #motivation
+
+This RFC addresses two problems with the way we move versions around in qri:
+1. datasets published to the registry are creating weird states where users can't get versions
+2. the terms we use for moving versions around are confusing.
+
+The first problem comes down to improper defaults. Publish currently sends _only the latest version_ of a dataset. If a user creates five versions, then runs publish, only the fifth version is pushed, leaving any party intersted in earlier versions with only the author as a viable source of versioned data. If the author isn't online and dialable, those prior versions are unavailable to the network. On the upside, our work on logbook and log syncronization means that at least the _history_ of the dataset is visible.
+
+Publishing should default to pushing _as much data as the remote is willing to accept_. Publishers sending data to a willing host should be trying to keep the network as robust as possible by making lots of copies.
```

Comment:
I think we should elaborate on this a bit more. One of my concerns here is if you cannot fit the last X revisions. Do you send over the oldest Y versions or the newest Z. Also think this should potentially not be configurable on the user side and we should stick to 1 default otherwise it would end up just being erratic behavior for end users.

The resolution of this issue should be done on the recieving end ideally a "registry" supporting some sort of "version distribution" among peers to accomodate and spread out the data while for a "repo" it should be fine to outright reject requests based on its policy.